src/pages/[platform]/build-a-backend/storage/set-up-storage/index.mdx

```diff
@@ -534,25 +534,40 @@ upload.addEventListener("click", () => {
 <InlineFilter filters={["swift"]}>
 
 ```swift
-let dataString = "My Data"
-let fileName = "myPhoto.png"
-guard let fileUrl = FileManager.default.urls(
-    for: .documentDirectory,
-    in: .userDomainMask
-).first?.appendingPathComponent(fileName)
-else { return }
-
-try dataString.write(
-    to: fileUrl,
-    atomically: true,
-    encoding: .utf8
-)
-
-let uploadTask = Amplify.Storage.uploadFile(
-    path: .fromString("picture-submissions/myPhoto.png"),
-    local: filename
-)
+import Amplify
+import SwiftUI
+import PhotosUI
+
+struct ContentView: View {
+    @State private var selectedPhoto: PhotosPickerItem?
+    @State private var image: Image?
+
+    var body: some View {
+        NavigationStack {
+            VStack {
+                image?
+                    .resizable()
+                    .scaledToFit()
+            }
+            .padding()
+            PhotosPicker(
+                selection: $selectedPhoto
+            ) {
+                Text("Select a photo to upload")
+            }
+            .task(id: selectedPhoto) {
+                image = try? await selectedPhoto?.loadTransferable(type: Image.self)
+                if let imageData = try? await selectedPhoto?.loadTransferable(type: Data.self) {
+                    let uploadTask = Amplify.Storage.uploadData(
+                        path: .fromString("picture-submissions/myPhoto.png"),
+                        data: imageData
+                    )
```

Comment:
should we show the rest of the upload, such as progress?

```swift
Task {
    for await progress in await uploadTask.progress {
        print("Progress: \(progress)")
    }
}
let value = try await uploadTask.value
print("Completed: \(value)")
```
(https://docs.amplify.aws/swift/build-a-backend/storage/upload/#upload-data)

Reply:
I think it's not needed in the set up stage. We just want to show an example of how they would do it